addons/docs/docs/mdx.md

```diff
@@ -0,0 +1,180 @@
+<center>
+  <img src="./media/hero.png" width="100%" />
+</center>
+
+# Storybook Docs MDX
+
+> ⚠️ MDX support is an experimental feature in Storybook 5.2. The API may change in 5.3 outside of the normal semver rules. Be forewarned!
+
+`MDX` is the syntax [Storybook Docs](../README.md) uses to capture long-form markdown documentation and stories in one file. You can also write pure documentation pages in `MDX` and add them to Storybook alongside your stories.
+
+- [Basic example](#basic-example)
+- [MDX-Flavored CSF](#mdx-flavored-csf)
+- [Writing stories](#writing-stories)
+- [Embedding stories](#embedding-stories)
+- [Decorators and parameters](#decorators-and-parameters)
+- [More resources](#more-resources)
+
+## Basic example
+
+Let's get started with a simple example that combines markdown with a single story:
+
+```md
+import { Meta, Story, Preview } from '@storybook/addon-docs/blocks';
+import { Checkbox } from './Checkbox';
+
+<Meta title="MDX|Checkbox" component={Checkbox} />
+
+# Checkbox
+
+With `MDX` we can define a story for `Checkbox` right in the middle of our
+markdown documentation.
+
+<Preview>
+  <Story name="all checkboxes">
+    <form>
+      <Checkbox id="Unchecked" label="Unchecked" />
+      <Checkbox id="Checked" label="Checked" checked />
+      <Checkbox appearance="secondary" id="second" label="Secondary" checked />
+    </form>
+  </Story>
+</Preview>
+```
+
+And here's how that's rendered in Storybook:
+
+<center>
+  <img src="./media/mdx-simple.png" width="100%" />
+</center>
+
+As you can see there's a lot going on here. We're writing Markdown, we're writing JSX, and somehow we're also defining Storybook stories that are drop-in compatible with the entire Storybook ecosystem.
+
+Let's break it down.
+
+## MDX-Flavored CSF
+
+[MDX](https://mdxjs.com/) is a standard file format that combines Markdown with JSX. This means you can use Markdown’s terse syntax (such as `# heading`) for your documentation, and freely embed JSX component blocks at any point in the file.
+
+MDX-Flavored CSF includes a collection of components called **"Doc Blocks"**, that allow Storybook to translate MDX files into storybook stories. MDX-defined stories are identical to regular Storybook stories, so they can be used with Storybook's entire ecosystem of addons and view layers.
```

Comment:
This is the first time the acronym is used in this doc. I think it'd be easier to understand if CSF was unpacked to `[Component Story Format (CSF)](https://medium.com/storybookjs/component-story-format-66f4c32366df)` here, and in the next paragraph abbreviated to CSF.